crates/cli/src/detect.rs

```diff
@@ -0,0 +1,68 @@
+use duct::cmd;
+use std::io;
+use std::path::Path;
+
+/// Find an executable in the `PATH`.
+pub(crate) fn find_executable(exe_name: impl AsRef<Path>) -> Option<std::path::PathBuf> {
+    std::env::var_os("PATH").and_then(|paths| {
+        std::env::split_paths(&paths)
+            .filter_map(|dir| {
+                let full_path = dir.join(&exe_name);
+                full_path.is_file().then_some(full_path)
+            })
+            .next()
```

Comment:
Nit: would be slightly cleaner IMO

```suggestion
            .map(|dir| dir.join(&exe_name))
            .find(Path::is_file)
```